src/csharp/Grpc.Tools/DepFileUtil.cs

```diff
@@ -158,21 +176,45 @@ internal static class DepFileUtil
         /// project and solution directories, which are also some level deep from the root.
         /// Instead of creating long and unwieldy names for these proto sources, we cache
         /// the full path of the name without the filename, and append the filename to it,
-        /// as in e. g. "foo/file.proto" will yield the name "deadbeef12345678_file", where
-        /// "deadbeef12345678" is a presumed hash value of the string "foo/". This allows
+        /// as in e. g. "foo/file.proto" will yield the name "deadbeef12345678/file", where
+        /// "deadbeef12345678" is a presumed hash value of the string "foo". This allows
         /// the file names be short, unique (up to a hash collision), and still allowing
         /// the user to guess their provenance.
         /// </remarks>
         public static string GetDepFilenameForProto(string protoDepDir, string proto)
         {
-            string dirname = Path.GetDirectoryName(proto);
-            if (Platform.IsFsCaseInsensitive)
-            {
-                dirname = dirname.ToLowerInvariant();
-            }
-            string dirhash = HashString64Hex(dirname);
+            string outdir = GetOutputDirWithHash(protoDepDir, proto);
             string filename = Path.GetFileNameWithoutExtension(proto);
-            return Path.Combine(protoDepDir, $"{dirhash}_{filename}.protodep");
+            return Path.Combine(outdir, $"{filename}.protodep");
+        }
+
+        /// <summary>
+        /// Construct relative output directory with directory hash
+        /// </summary>
+        /// <param name="outputDir">Relative path to the output directory, e. g. "out"</param>
+        /// <param name="proto">Relative path to the proto item, e. g. "foo/file.proto"</param>
+        /// <returns>
+        /// Full relative path to the directory, e. g. "out/deadbeef12345678"
+        /// </returns>
+        /// <remarks>
+        /// Since a project may contain proto files with the same filename but in different
```

Comment:
I really like the detailed explanation of why to hash the directory paths, but right now there's basically two copies of the same comment - perhaps we could update the xmldoc of GetDepFilenameForProto to just "see details on directory path hashing in GetOutputDirWithHash", instead of copy pasting.

Reply:
You are right. I first had it explained separately but in the end also moved the deps file into that folder instead of having the hash as part of the file name. Will resolve it :)